googledns/client.go

```diff
@@ -0,0 +1,137 @@
+package googledns
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/libdns/libdns"
+	"google.golang.org/api/dns/v1"
+	"google.golang.org/api/googleapi"
+	"google.golang.org/api/option"
+)
+
+const (
+	// zoneMapTTL timeout for the Google Cloud DNS zone map
+	zoneMapTTL = time.Minute * 5
+)
+
+// NewService initializes the Google client for the provider using the specified JSON file for credentials if set.
+func (p *Provider) NewService(ctx context.Context) error {
```

Comment:
Unfortunately, this will be a deal-breaker for inclusion in the libdns repositories; one of the [requirements](https://github.com/libdns/libdns/wiki/Implementing-providers) is that no additional provisioning steps are necessary:

> Explicit/exported provisioning steps should be avoided. They break a provider's drop-in-replacement-ability with other providers because the user has to know to first provision certain types before using them. Instead of requiring explicit provisioning, try to do it implicitly in each exported method that relies on it. If provisioning is expensive, it can be done just once by storing it in the struct (this requires synchronization for thread-safety) or by using sync.Once (which does the synchronization for you) the first time it is needed.

I think a few other implementations just do it implicitly on first use.

Reply:
Understood 👍 . I noticed other packages were using `sync.Mutex` so It shouldn't be difficult matching that pattern. Let me get that change in and give it a quick test.

Reply:
Yep, sure; or `sync.Once()` if you prefer.

Reply:
The service provisions on first use without having to call `newService` so we should be good to go there.